woocommerce/woocommerce-accommodation-bookings
# Set up e2e tests utilizing Playwright

### All Submissions:


### Changes proposed in this Pull Request:
PR adds following E2E tests using Playwright and E2E tests workflow which runs based on conditions.


| Area | Flow Name | Status |
| -- | -- | -- |
| Installation | Verify Plugin Activation Without Errors | ✅ |
| Installation | Verify can see a message when the core plugins WooCommerce & WooCommerce Booking are not active | ✅ |
| Setup | Verify Plugin Setup and Configuration Screen  | ✅ |
| Product | Can create an Accommodation product | ✅ |
| Product | Verify labels 'Select check-in' and 'Select check-out' works expected.| ✅ |
| Product | Verify Setting Minimum Nights Restriction - "Minimum number of nights allowed in a booking" | ✅ |
| Product | Verify Setting Maximum Nights Restriction "Maximum number of nights allowed in a booking" | ✅ |
| Product | Verify the Number of Rooms Availability | ✅ |
| Product | Verify Calendar DisplayMode - Display Calendar on Click | ✅ |
| Product | Verify Calendar Display Mode Setting - Always Visible | ✅ |
| Product | Verify Accommodation Booking Creation with Confirmation Setting | ✅ |
| Product | Verify Accommodation Booking Process for Slot Requiring Confirmation | ✅ |
| Product | Check for "Can be cancelled" Setting | ✅ |
| Product | Availability > Bookings Can Be Made Starting - Month/Week/Day - in the Future Setting | Partial - one case failing due to bug |
| Product | Availability > Booking Can Be Booked Till Setting  |  Partial - one case failing due to bug |
| Product | Verify Rate Setting should work as expected | ✅ |
| Product | Verify 'Rates' Setting for Accommodation Booking with Range Types with Range Type 'Range of certain night'. | ✅ |
| Product | Verify 'Rates' Setting for Accommodation Booking with Range Types with Range Type 'Range of months'. | ✅ |
| Order Flow | Verify Booking Details Check-In and Check-Out Date and Time Display. | ✅ |
| Order and emails | Verify Accommodation Booking Details in Confirmation Emails | ✅ |



- Condition-based GH action run
	- PR must have `needs: e2e testing` to run it.
	- `status: e2e tests passing` or `status: e2e tests failing` label will be added to the PR to indicate the status of the E2E test result.
	- Push to the `smoke-testing` branch will run all e2e tests(smoke testing)

Closes #354

### How to test the changes in this Pull Request:

**Test this Locally:**
1. Checkout to PR
2. Run `npm install`
3. Run `npx playwright install`
4. Run `BOT_GT_TOKEN=xxx npm run env:install-plugins`
5. Run `npm run env:start` (Note: Please start docker before this command)
7. Run `npm run test:e2e`
8. Verify all tests run successfully.

**Test In this PR**
1. Add the `needs: e2e testing` label to PR.
2. Adding a label on PR will start the E2E test job and it will run "E2E tests" against PR.
3. Verify that Tests pass successfully, the `needs: e2e testing` label is removed, and the `status: e2e tests passing` label added to the PR

### Other information:




### Changelog entry
> Dev - Add end-to-end tests using Playwright.
